server/routes/courses.js

```diff
@@ -8,7 +8,13 @@ let CourseOffering = require('../models/courseOffering');
 //     .catch(err => res.status(400).json('Error: ' + err));
 // });
 
-router.get('/getSingle/:courseId', function(req, res, next) {
```

Comment:
I think this order problem is why it was helpful to have the additional '/getSingle' specification for getting a course by id in the path name.
We could also pass the id in the body of the request, and avoid the :courseId param altogether

Reply:
i mean yea i was thinking that too but idk it feel like it just feels right to make the get-by-id endpoint be `/courses/:courseId`, so i  was thinking we just keep the get endpoint at the bottom and keep it

Reply:
Hmm if so, you should explicitly document this in the code, cause it's impossible to know this by just reading the route definitions.